web-app/src/components/BookingModal/index.js

```diff
@@ -1,102 +1,131 @@
 import React from 'react';
-import DatePicker from 'react-datepicker';
 import { PropTypes as PT } from 'prop-types';
-import { Button, Modal } from '../common';
+import { Button, Modal, Input } from '../common';
 import FontAwesomeIcon from '@fortawesome/react-fontawesome';
 import { faTimes } from '@fortawesome/fontawesome-free-solid';
-import { StyleContainer, BookingStatus, BookingInputs, StatusDot, ButtonWrap } from './styled';
+import {
+  StyleContainer,
+  BookingStatus,
+  StatusDot,
+  ButtonWrap,
+  Form,
+} from './styled';
 import { statusText } from '../../utilities/holidayStatus';
 
 const BookingModal = props => {
   const {
     closeModal,
     showModal,
     booking,
-    changeStart,
-    changeEnd,
-    changeHalfday,
+    formElementsArray,
     requestHoliday,
     cancelHoliday,
     updateHoliday,
-    changeWFH,
+    formChanged,
   } = props;
 
-  const getActions = () => (
-    booking.title ?
+  const getActions = () =>
+    booking.title ? (
       <ButtonWrap>
-        <Button id="updateHolidayBtn" onClick={updateHoliday} label={'Update'} />
-        <Button id="cancelHolidayBtn" onClick={cancelHoliday} label={'Cancel Booking'} />
+        <Button
+          id="updateHolidayBtn"
+          onClick={updateHoliday}
+          label={'Update'}
+          disabled={!booking.formIsValid}
+        />
+        <Button
+          id="cancelHolidayBtn"
+          onClick={cancelHoliday}
+          label={'Cancel Booking'}
+          disabled={!booking.formIsValid}
+        />
       </ButtonWrap>
-      :
-      <Button id="requestHolidayBtn" onClick={requestHoliday} label={booking.isWFH ? 'Add' : 'Request'} />
-  );
+    ) : (
+      <Button
+        id="requestHolidayBtn"
+        onClick={requestHoliday}
+        label={booking.form.isWFH.value ? 'Add' : 'Request'}
+        disabled={!booking.formIsValid}
+      />
+    );
+
+  // const getTotalDays = () => {
+  //   if (booking.form.isWFH.value) {
+  //     return '0';
+  //   }
+  //   return (
+  //     Math.floor(booking.duration) / (booking.form.isHalfday.value ? 2 : 1)
+  //   );
+  // };
```

Comment:
Remove unused code

Reply:
I kept this commented as a why to remind me to ask you what it was for. The value that this function returns is already being handled in and contained booking.duration? But yes, we can talk about this tomorrow.